### DISCUSSION TITLE
how to install manually on windows with no installer

### DISCUSSION THREAD

**Discussion by NatchTheTree** [openssl/openssl#24449] [Q&A]
im asking because i have looked everywhere in this forum but i cant find a step by step guide how to get openssl running from tar.gz file to being able to check the openssl version on powershell.
i downloaded my file from the official site (tar.gz). i unpacked them and found out that there are no installer. i tried following guides i found on google but they didnt work. how do i get it installed?
i know that there are other option like shining light production that has an .exe installer but i still want to know how to set it up using the official file from the official site.

**Comment by mattcaswell** [comment]
OpenSSL is distributed in a source only format. This means you have to build it yourself before it can be installed.

There are a number of third parties that distribute pre-built binaries which are good options for most people. We maintain a list of them here:

https://wiki.openssl.org/index.php/Binaries

If you really want to build it yourself then you should extract the tar.gz archive somewhere and look at the INSTALL.md file that is at the top of the archive:

https://github.com/openssl/openssl/blob/openssl-3.3/INSTALL.md

Read this file carefully. It gives the installation instructions for all platforms including Windows.

There are also some Windows specific notes here:

https://github.com/openssl/openssl/blob/openssl-3.3/NOTES-WINDOWS.md

This file gives some useful links and information on how to get hold of the prerequisites required to build OpenSSL. I recommend creating a native build using Visual C++. You will need to install Visual C++ first, along with perl and nasm. There is also a useful Windows specific quick-start guide in this file.

**Comment by nhorman** [comment]
If it helps, we are actively building an installer, that is available here:
https://github.com/openssl/installer
It requires following the above steps @mattcaswell has referenced, after which you can use the nsis script in the aforementioned repository to create an unsigned installer binary
